apps/src/applab/RecordListener.js

```diff
@@ -0,0 +1,230 @@
+'use strict';
+
+/**
+ * Class which allows callers to listen to changes to data tables. This is currently
```

Comment:
What is the rollout plan for the listener function?  I think we probably wouldn't want to make the polling-based implementation generally available;  as we saw from Netsim even a moderate size class doing coordinated polling and fetching is enough to pose a scale risk to our servers.  Rather, we can make it available on a limited basis to get feedback on the API.